### DISCUSSION TITLE
OAuth2PasswordRequestForm returns 422 instead of 400

### DISCUSSION THREAD

**Discussion by retnikt** [fastapi/fastapi#9056] [Questions]
### Describe the bug

When a validation error occurs on an OAuth2PasswordReqeustForm, the HTTP response code is 422 Unprocessable Entity). However, the [OAuth2 spec](https://tools.ietf.org/html/rfc6749#section-5.2) specifies that either a 400 Bad Request or 401 Unauthorized MUST be returned. Also, a generic pydantic error JSON is returned, but the RFC specifies differently. Therefore FastAPI's imlementation is non-conformant. I understand these issues can be worked around in a custom implementation, but I feel that the default, batteries-included implementation should follow the spec correctly. Arguably a [422 is actually more suitable](https://tools.ietf.org/html/rfc4918#section-11.2) in this case, but its use is also likely to break OAuth2 client libraries.

### To Reproduce

1. Create a file with:

```Python
from fastapi import FastAPI
from fastapi.security.oauth2 import OAuth2PasswordBearer, OAuth2PasswordRequestForm

app = FastAPI()
oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/token")

@app.post("/token")
def login(form: OAuth2PasswordRequestForm):
    return {"access_token": "faketoken", "token_type": "bearer"}
```

3. Send an invalid request form
4. Observe the non-conformant 422 error with pydantic JSON error response

### Expected behavior

A 400 error is returned with a valid error response as described in https://tools.ietf.org/html/rfc6749#section-5.2

**Comment by andrew222651** [comment]
Not sure there's an ideal way to implement this without a significant architectural change.

**Comment by stephenoneal** [comment]
422 is a webdav thing.  a lot of APIs may choose to leverage this explicitly but it seems odd that this is a default unless this was settled upon by digging through stackoverflow comments.  I could be wrong, but I haven't seen an RFC that indicates that 422 is the valid for any response except a webdav interface so it seems odd that fastapi defaults to it.

**Comment by sarathak** [comment]
422 is Unprocessable Entity

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

**Comment by RafaelWO** [comment]
Just in case someone else came across here but actually has made a mistake in terms of routing priority. My issue of an HTTP 422 instead of 401 is related to #1256
